### DISCUSSION TITLE
docker build errors suddenly

### DISCUSSION THREAD

**Discussion by earlsioson** [Kitware/trame#299] [Q&A]
After rebuilding our docker images we started receiving these errors now...

Starting server...
/opt/trame/activate_venv.sh: line 1: /deploy/server/venv/bin/activate: No such file or directory
/opt/trame/run.sh: line 20: /deploy/server/launcher.json: No such file or directory
Starting the wslink launcher at
/usr/bin/python: Error while finding module specification for 'wslink.launcher' (ModuleNotFoundError: No module named 'wslink')

These docker containers were created via the cookie cutter app and were working previously. This only started happening after rebuilding the docker image. Did something change in the base image that caused this error?

**Comment by jourdain** [comment]
Maybe the pull of trame v3 if you didn't list all your dependencies? 
But the log of the build steps of your image should reveal the source of the problem and why the venv creation failed.

**Reply by earlsioson** [reply]
@jourdain I pinned the trame version in setup.cfg to trame==2.5.2 and was able to build the image succesfully and the venv gets created in server/venv now, but when I run the image we get this in the log...

docker logs -f 709d5bb61907
 * Restarting Apache httpd web server apache2
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.3. Set the 'ServerName' directive globally to suppress this message
   ...done.
Starting the wslink launcher at

I tried to check the logs in server/logs dir but the apache/access.log and apache/error.log are empty and the launcher log directory is also empty. How can I troubleshoot this further?

**Reply by jourdain** [reply]
The listed error message is expected and does not affect the service. And based on that output, it should be working just fine.

Do you still get the same error as initially? If so, did you remove the server directory before building your image again?

**Reply by earlsioson** [reply]
@jourdain No I do not get the same error as initially. Yes I removed the server directory. All that I see is this now with a session error in the network tab...
<img width="1767" alt="image" src="https://github.com/Kitware/trame/assets/13466524/b4a96300-fd45-4090-bb09-2980c5511a5f">

**Reply by earlsioson** [reply]
There are no logs being written to in the server logs directory.

**Reply by jourdain** [reply]
Anything on the session log output? (in /deploy/server/logs/launcher/)

**Reply by earlsioson** [reply]
root@59b4e651f7b8:/# cat /deploy/server/logs/launcher/5b866e38-3151-11ee-a198-0242ac110002.txt
ParaView is using venv: /deploy/server/venv
Traceback (most recent call last):
  File "/opt/paraview/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/paraview/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/paraview/lib/python3.9/site-packages/paraview/apps/trame.py", line 26, in <module>
    main()
  File "/opt/paraview/lib/python3.9/site-packages/paraview/apps/trame.py", line 22, in main
    module = importlib.import_module(TRAME_APP)
  File "/opt/paraview/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 855, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/deploy/server/venv/lib/python3.9/site-packages/tbreak/__init__.py", line 1, in <module>
    from .app import main
  File "/deploy/server/venv/lib/python3.9/site-packages/tbreak/app/__init__.py", line 1, in <module>
    from .main import main
  File "/deploy/server/venv/lib/python3.9/site-packages/tbreak/app/main.py", line 13, in <module>
    from . import engine, ui
  File "/deploy/server/venv/lib/python3.9/site-packages/tbreak/app/engine.py", line 12, in <module>
    from vtk.util import numpy_support
  File "/deploy/server/venv/lib/python3.9/site-packages/vtk.py", line 29, in <module>
    from vtkmodules.vtkViewsInfovis import *
ModuleNotFoundError: No module named 'vtkmodules.vtkViewsInfovis'

**Reply by jourdain** [reply]
This is really weird. Did you try the plain cookiecutter to check if the plain app is working? If not, I can double check it.

**Reply by jourdain** [reply]
Oh, so it seems to be an issue with your vtk package.

**Comment by earlsioson** [comment]
The above error is for a trame v2.5.2. which I pinned trame==2.5.2. That's how I got rid if the initial error.

As far as the cookiecutter app. I did try it with trame v3 and it initially did not until I updated the setup.cfg with the necessary new packages required for trame v3

**Reply by jourdain** [reply]
Normally the latest version of the cookiecutter add those missing dependencies to properly handle v3.

**Comment by earlsioson** [comment]
We ended up upgrading our repos to trame v3 to resolve this.

**Reply by jourdain** [reply]
Do you know what was the deal?

**Reply by earlsioson** [reply]
Unfortunately we couldn't determine why rebuilding our trame v2 images broke. We tried using trame==2.5.2 in the setup.cfg but still received the VTK module error. Since it was easier to update to trame v3 we decided to upgrade rather than try and figure out what broke in our trame v2 build.

We are now versioning and tagging our docker images, so we don't have to rebuild the images in other environments.

**Reply by jourdain** [reply]
The trame docker images are also tagged monthly if that helps.
